examples/run/cpu/misaligned_truth_fitting_example.cpp

```diff
@@ -0,0 +1,219 @@
+#include <detray/propagator/rk_stepper.hpp>
+
+// VecMem include(s).
+#include <vecmem/memory/host_memory_resource.hpp>
+#include <vecmem/utils/copy.hpp>
+
+// System include(s).
+#include <cstdlib>
+#include <exception>
+#include <iomanip>
+#include <iostream>
+
+using namespace traccc;
+namespace po = boost::program_options;
+
+// The main routine
+//
+int main(int argc, char* argv[]) {
+    std::unique_ptr<const traccc::Logger> ilogger = traccc::getDefaultLogger(
+        "TracccExampleTruthFitting", traccc::Logging::Level::INFO);
+
+    TRACCC_LOCAL_LOGGER(std::move(ilogger));
+
+    // Program options.
+    traccc::opts::detector detector_opts;
+    traccc::opts::input_data input_opts;
+    traccc::opts::track_propagation propagation_opts;
+    traccc::opts::track_fitting fitting_opts;
+    traccc::opts::performance performance_opts;
+    traccc::opts::program_options program_opts{
+        "Truth Track Fitting on the Host",
+        {detector_opts, input_opts, propagation_opts, fitting_opts,
+         performance_opts},
+        argc,
+        argv,
+        logger().cloneWithSuffix("Options")};
+
+    /// Type declarations
+    using host_detector_type = traccc::default_detector::host;
+
+    // Memory resources used by the application.
+    vecmem::host_memory_resource host_mr;
+    // Copy obejct
+    vecmem::copy copy;
+
+    // Performance writer
+    traccc::fitting_performance_writer fit_performance_writer(
+        traccc::fitting_performance_writer::config{});
+
+    /*****************************
+     * Build a geometry
+     *****************************/
+
+    // B field value and its type
+    // @TODO: Set B field as argument
+    const traccc::vector3 B{0, 0, 2 * traccc::unit<traccc::scalar>::T};
+    auto field = detray::bfield::create_const_field<traccc::scalar>(B);
+
+    // Read the detector
+    detray::io::detector_reader_config reader_cfg{};
+    reader_cfg.add_file(
+        traccc::io::get_absolute_path(detector_opts.detector_file));
+    if (!detector_opts.material_file.empty()) {
+        reader_cfg.add_file(
+            traccc::io::get_absolute_path(detector_opts.material_file));
+    }
+    if (!detector_opts.grid_file.empty()) {
+        reader_cfg.add_file(
+            traccc::io::get_absolute_path(detector_opts.grid_file));
+    }
+    const auto [host_det, names] =
+        detray::io::read_detector<host_detector_type>(host_mr, reader_cfg);
+
+    /// Create a "misaligned" context in the transform store
+    using xf_container = host_detector_type::transform_container;
+    using xf_vector = xf_container::base_type;
+
+    const xf_container& default_xfs = host_det.transform_store();
+    xf_vector misaligned_xfs;
+    misaligned_xfs.reserve(default_xfs.size());
+    for (const auto& xf : default_xfs) {
+        misaligned_xfs.push_back(xf);
+    }
+    xf_container* ptr_default_xfs = const_cast<xf_container*>(&default_xfs);
+    ptr_default_xfs->add_context(misaligned_xfs);
```

Comment:
I think for now this is fine, but let me consider how we can do this without a const cast

Reply:
Yes, it would be good to move away from the `const_cast` eventually. Thanks